File: Tool/template/java/Template1.java
package TableConfig;
import java.util.HashMap;
import java.util.Map;
import java.util.List;
import java.util.ArrayList;

public class TemplateClass extends DataBase
{
	//region value
ValueContent
	//endregion
    //region load and get funtion
	private static TemplateClass[] m_Datas = null;
	public static TemplateClass[] GetDatas()
	{
		return m_Datas;
	}

	
	public static int GetCount()
	{
		if(m_Datas == null)
		{
			return 0;
		}
		return m_Datas.length;
	}


	private static Map<FirstKeyType,TemplateClass> m_DicDatas = null;
	public static Map<FirstKeyType, TemplateClass> GetDicDatas()
	{
		return m_DicDatas;
	}
	


	private void Load(BinData bin)
	{
ReadValueLoadContent		
	}

	public static void Load()
	{
		BinData bin = OpenData("TemplateClass.bin")	;
		if (bin != null)
		{
			try
			{
				int count = bin.ReadInt32();
				m_Datas = new TemplateClass[count];
				m_DicDatas = new HashMap<>();
				for (int i = 0; i < count; ++i)
				{
					TemplateClass data = new TemplateClass();
					data.Load(bin);
					m_Datas[i] = data;
					m_DicDatas.put(data.GetFirstKey(), data);
				}
			}
			catch(Exception e)
			{
				
			}
		}
	}



	public static TemplateClass Get(FirstKeyType FirstKey)
	{
		TemplateClass data = m_DicDatas.get(FirstKey);
		return data;
	}



	public static void Reload()
	{
		Unload();
		Load();
	}

	public static void Unload()
	{
		boolean bGC = false;
		if(m_Datas != null)
		{
			m_Datas = null;
			bGC = true;
		}
		if(bGC)
		{
			System.gc();
		}
	}
	//endregion
}

File: Tool/template/c++/Template1.cpp
#include "TemplateClass.h"

std::vector<TemplateClass*>* TemplateClass::m_Datas = NULL;
std::map<FirstKeyType, TemplateClass*>* TemplateClass::m_DatasDic = NULL;

int TemplateClass::GetCount()
{
	if(m_Datas == NULL)
	{
		return NULL;
	}
    return m_Datas->size();
}

std::vector<TemplateClass*>* TemplateClass::GetDatas()
{
    return m_Datas;
}

TemplateClass* TemplateClass::Get(FirstKeyType FirstKey)
{
	if(m_DatasDic == NULL)
	{
		return NULL;
	}
	auto it = m_DatasDic->find(FirstKey);
	if (it != m_DatasDic->end()) {
		// 键存在于map中
		return it->second; // 访问键对应的值
	}
	else {
		return NULL;
	}
}

void TemplateClass::Load(BinData* bin)
{
ReadValueLoadContent	
}



void TemplateClass::Load() 
{
	BinData* bin = OpenBin("TemplateClass.bin");
	if (bin != NULL && bin->IsOpen())
	{
		try
		{
			int count = bin->ReadInt32();
			m_Datas = new std::vector<TemplateClass*>(count);
			m_DatasDic = new std::map<FirstKeyType, TemplateClass*>();
			for (int i = 0; i < count; ++i)
			{
				TemplateClass * data = new TemplateClass();
				data->Load(bin);
				m_Datas->push_back(data);
				if(Get(data->GetFirstKey()) == NULL)
				{
					m_DatasDic->insert(std::make_pair(data->GetFirstKey(), data));
				}
			}
		}
		catch (const std::exception& e)
		{

		}
		bin->Close();
		delete bin;
		bin = NULL;
	}
}

void TemplateClass::Reload()
{
    Unload();
    TemplateClass::Load();
}

void TemplateClass::Unload()
{
   if (m_Datas == NULL)
	{
		return;
	}

	for (std::vector<TemplateClass*>::iterator it = m_Datas->begin(); it != m_Datas->end(); it++) {
		delete *it;
	}

	m_Datas->clear();
	m_DatasDic->clear();
	delete m_Datas;
	delete m_DatasDic;
	m_Datas = NULL;
	m_DatasDic = NULL;
}


